### PR TITLE
feat: :construction: Add aeSwap features directly in aeWallet

### LINKED ISSUE #1085: Add DEX features
### Is your feature request related to a problem?

_No response_

### Describe the solution you'd like

Add swap and earn processes in aeWallet
### Additional context

_No response_

### Epic

_No response_

### PR DESCRIPTION
**Pull request by redDwarf03**
# Description

Fixes #1085 

## Type of change

Please delete options that are not relevant.
- New feature (non-breaking change which adds functionality)
- Refactorization

## Material used:
- iOS (Smartphone/Tablet)
- Android (Smartphone/Tablet)
- Browser
- Linux
- Windows
- iOS (Mac)

## Checklist:

- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: lib/domain/usecases/aeswap/swap.usecase.dart [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+  final archethic.ApiService apiService;
+  final ns.TaskNotificationService<DexNotification, aedappfm.Failure>
+      notificationService;
+  final aedappfm.VerifiedTokensRepositoryInterface verifiedTokensRepository;
+  final TransactionRemoteRepositoryInterface transactionRepository;
+  final KeychainSecuredInfos keychainSecuredInfos;
+  final Account selectedAccount;
+
+  Future<void> run(
+    Ref ref, // TODO(Chralu): usecases should not depend on riverpod
```
Comment (Chralu) [inline lib/domain/usecases/aeswap/swap.usecase.dart]: Passer directement la balance des 2 tokens en parametre a la place du `Ref`.

File: lib/infrastructure/repositories/transaction/archethic_transaction.dart [hunk 2]
```diff
@@ -232,6 +232,34 @@ class ArchethicTransactionRepository
     );
   }
 
+  Future<archethic.Transaction> buildTransactionRaw(
```
Comment (Chralu) [inline lib/infrastructure/repositories/transaction/archethic_transaction.dart:236]: Add `@override` annotation

File: lib/router/router.dart [hunk 3]
```diff
@@ -92,6 +116,7 @@ class RoutesPath {
             ),
             ..._authenticationRoutes,
             ..._introductionRoutes,
+            ..._aeSwapRoutes,
```
Comment (Chralu) [inline lib/router/router.dart]: Mettre ces routes sous le `AutoLockGuard` (ligne 177) afin d'avoir le lock screen dessus :

```dart
routes: [
    ..._authenticatedRoutes,
    ..._aeSwapRoutes,
]
```

File: lib/ui/views/add_account/bloc/provider.dart [hunk 4]
```diff
@@ -192,7 +190,8 @@ class AddAccountFormNotifier extends AutoDisposeNotifier<AddAccountFormState> {
   }
 
   Future<void> removeAccount(BuildContext context, String account) async {
-    final transactionRepository = ref.read(AddAccountFormProvider._repository);
+    final transactionRepository =
```
Comment (Chralu) [inline lib/ui/views/add_account/bloc/provider.dart:193]: Pas de watch dans une action.

File: lib/ui/views/main/components/main_appbar.dart [hunk 5]
```diff
@@ -86,21 +87,47 @@ class MainAppBar extends ConsumerWidget implements PreferredSizeWidget {
...
           : preferences.mainScreenCurrentPage == 1
               ? _transactionsTab(context, ref)
-              : FittedBox(
-                  fit: BoxFit.fitWidth,
-                  child: AutoSizeText(
-                    localizations.aeSwapHeader,
-                    style: ArchethicThemeStyles.textStyleSize24W700Primary,
-                  ),
-                ).animate().fade(duration: const Duration(milliseconds: 300)),
+              : preferences.mainScreenCurrentPage == 2
```
Comment (Chralu) [inline lib/ui/views/main/components/main_appbar.dart:100]: Cette classe est difficile a lire. Mais bon tant pis.

File: lib/ui/views/main/components/menu_widget_wallet.dart [hunk 6]
```diff
@@ -134,7 +133,7 @@ class MenuWidgetWallet extends ConsumerWidget {
                         FeedbackType.light,
                         preferences.activeVibrations,
                       );
-                  context.go(BuySheet.routerPage);
+                  context.push(BuySheet.routerPage);
```
Comment (Chralu) [inline lib/ui/views/main/components/menu_widget_wallet.dart:136]: Verifier si le lockscreen marche encore par dessus ce menu

File: lib/ui/views/sheets/aeBridge_sheet.dart [hunk 7]
```diff
@@ -28,14 +26,14 @@ class DEXSheet extends ConsumerStatefulWidget {
...
 
   @override
-  ConsumerState<DEXSheet> createState() => DEXSheetState();
+  ConsumerState<AEBridgeSheet> createState() => AEBridgeSheetState();
 }
 
-class DEXSheetState extends ConsumerState<DEXSheet> {
-  String? aeSwapUrl;
+class AEBridgeSheetState extends ConsumerState<AEBridgeSheet> {
+  String? aeSwapUrl = 'https://bridge.testnet.archethic.net/bridge';
```
Comment (Chralu) [inline lib/ui/views/sheets/aeBridge_sheet.dart]: On ne devrait pas remettre la remote conf ici ?

File: lib/ui/views/sheets/buy_sheet.dart [hunk 8]
```diff
@@ -42,7 +43,7 @@ class BuySheet extends ConsumerWidget implements SheetSkeletonInterface {
         key: const Key('back'),
         color: ArchethicTheme.text,
         onPressed: () {
-          context.go(HomePage.routerPage);
+          context.pop();
```
Comment (Chralu) [inline lib/ui/views/sheets/buy_sheet.dart:46]: Je ne suis pas sur que ce soit une bonne idée de ne pas passer par GoRouter.

Si c'est vraiment ce qu'on veut, il faut supprimer la route de la config GoRouter.

File: lib/ui/views/tokens_detail/layouts/components/token_detail_menu.dart [hunk 9]
```diff
@@ -132,7 +135,41 @@ class TokenDetailMenu extends ConsumerWidget {
...
+                    text: localizations.swapHeader,
+                    icon: aedappfm.Iconsax.arrange_circle_2,
+                    onTap: () async {
+                      sl.get<HapticUtil>().feedback(
+                            FeedbackType.light,
+                            preferences.activeVibrations,
+                          );
+
+                      final params = {
+                        'from': aeToken.isUCO ? 'UCO' : aeToken.address,
```
Comment (Chralu) [inline lib/ui/views/tokens_detail/layouts/components/token_detail_menu.dart:150]: Je crois qu'on a desormais une constante pour l'addresse de token `UCO`.

File: lib/ui/views/tokens_detail/layouts/components/token_detail_menu.dart [hunk 10]
```diff
@@ -174,39 +211,33 @@ class TokenDetailMenu extends ConsumerWidget {
...
-                                    (settings) => settings.network,
-                                  ),
+                                final poolJson = jsonEncode(pool!.toJson());
+                                final poolEncoded =
+                                    Uri.encodeComponent(poolJson);
+                                final farmLockJson =
+                                    jsonEncode(farmLock!.toJson());
+                                final farmLockEncoded =
+                                    Uri.encodeComponent(farmLockJson);
+                                await context.push(
```
Comment (Chralu) [inline lib/ui/views/tokens_detail/layouts/components/token_detail_menu.dart:232]: Vérifier que le LockScreen fonctionne sur cette page `FarmLockDepositSheet`.

File: lib/ui/views/sheets/aeBridge_sheet.dart (hunk 7 above)
Comment (redDwarf03) [reply lib/ui/views/sheets/aeBridge_sheet.dart]: i did a static page to inform user about bridge access

File: lib/ui/views/tokens_detail/layouts/components/token_detail_menu.dart (hunk 9 above)
Comment (redDwarf03) [reply lib/ui/views/tokens_detail/layouts/components/token_detail_menu.dart:150]: not in the general aeToken management. some updates from aeSwap haven't been reported for the moment. We should work on it when we will create a common aeSwap SDK

**Pull request merged**